sensor/kubernetes/listener/resources/networkpolicy_test.go

```diff
@@ -0,0 +1,445 @@
+package resources
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/sensor/common/detector/mocks"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func TestGetSelector(t *testing.T) {
+	if !features.NetworkPolicySystemPolicy.Enabled() {
+		return
+	}
+
+	mockCtrl := gomock.NewController(t)
+	nps := newNetworkPoliciesStore()
+	ds := newDeploymentStore()
+	det := mocks.NewMockDetector(mockCtrl)
+
+	dispatcher := newNetworkPolicyDispatcher(nps, ds, det)
+
+	cases := []struct {
```

Comment:
nit: I think the tests are a bit easier to read if you define test cases as `map[string]struct{ ... }`.

Then you can write them as:
```go
"New NetworkPolicy": {
	netpol: &storage.NetworkPolicy{
		Id:        "1",
		Namespace: "default",
		Spec: &storage.NetworkPolicySpec{
			PodSelector: &storage.LabelSelector{
				MatchLabels: map[string]string{
					"app":  "sensor",
					"role": "backend",
				},
			},
		},
	},
	oldNetpol: nil,
	action:    central.ResourceAction_CREATE_RESOURCE,
	expectedSelector: []map[string]string{
		{
			"app":  "sensor",
			"role": "backend",
		},
	},
	expectedEmpty: false,
},
```

Reply:
Done. Thanks for the tip, it looks more readable this way.

Reply:
This is cool to use map instead of array! We actually loose the order of execution (it is random for map), but it doesn't play a role here.